new-packages/ts-project/__tests__/actions.test.ts

```diff
@@ -0,0 +1,691 @@
+    'tsconfig.json': JSON.stringify({}),
+    'index.ts': ts`
+      import { createMachine } from "xstate";
+
+      createMachine({
+        entry: [
+          {
+            type: "callDavid",
+          },
+        ],
+      });
+    `,
+  });
+
+  const project = await createTestProject(tmpPath);
+  expect(replaceUniqueIds(project.extractMachines('index.ts')))
+    .toMatchInlineSnapshot(`
+      [
+        [
+          {
+            "blocks": {
+              "block-0": {
+                "blockType": "action",
+                "parentId": "state-0",
+                "properties": {
+                  "params": {},
+                  "type": "callDavid",
+                },
+                "sourceId": "callDavid",
+                "uniqueId": "block-0",
+              },
+            },
+            "edges": {},
+            "implementations": {
+              "actions": {
+                "undefined": {
+                  "id": "callDavid",
+                  "name": "callDavid",
+                  "type": "action",
+                },
+              },
+              "actors": {},
+              "guards": {},
+            },
+            "nodes": {
+              "state-0": {
+                "data": {
+                  "description": undefined,
+                  "entry": [
+                    "block-0",
+                  ],
+                  "exit": [],
+                  "history": undefined,
+                  "initial": undefined,
+                  "invoke": [],
+                  "metaEntries": [],
+                  "tags": [],
+                  "type": "normal",
+                },
+                "parentId": undefined,
+                "type": "node",
+                "uniqueId": "state-0",
+              },
+            },
+            "root": "state-0",
+          },
+          [],
+        ],
+      ]
+    `);
+});
+
+test('should extract an inline action (function)', async () => {
+  const tmpPath = await testdir({
+    'tsconfig.json': JSON.stringify({}),
+    'index.ts': ts`
+      import { createMachine } from "xstate";
+
+      createMachine({
+        entry: () => {},
+      });
+    `,
+  });
+
+  const project = await createTestProject(tmpPath);
+  expect(replaceUniqueIds(project.extractMachines('index.ts')))
+    .toMatchInlineSnapshot(`
+      [
+        [
+          {
+            "blocks": {
+              "block-0": {
+                "blockType": "action",
+                "parentId": "state-0",
+                "properties": {
+                  "params": {},
+                  "type": "inline:action-0",
+                },
+                "sourceId": "inline:action-0",
+                "uniqueId": "block-0",
+              },
+            },
+            "edges": {},
+            "implementations": {
+              "actions": {
+                "inline:action-0": {
+                  "id": "inline:action-0",
+                  "name": "inline:action-0",
+                  "type": "action",
```

Comment:
The source must have a `code` property that contains the source code of the action `() => {}`

Reply:
How would it be used by the Studio?

Reply:
If we don't show the source panel and exported code, it won't be used at all.